velox/type/tests/DecimalTest.cpp

```diff
@@ -106,6 +106,22 @@ void testToByteArray(int128_t value, int8_t* expected, int32_t size) {
   EXPECT_EQ(std::memcmp(expected, out, length), 0);
 }
 
+template <typename T>
+void testConvertToStringView(
+    T unscaleValue,
+    int precision,
+    int scale,
+    int expectedMaxStringSize,
+    std::string expected) {
+  ASSERT_EQ(
+      DecimalUtil::maxStringViewSize(precision, scale), expectedMaxStringSize);
+  char out[expectedMaxStringSize];
+  DecimalUtil::convertToStringView<T>(
+      unscaleValue, scale, expectedMaxStringSize, out);
+  EXPECT_EQ(std::memcmp(expected.data(), out, expected.size()), 0)
+      << StringView(out, expectedMaxStringSize);
```

Comment:
What would the error message be like? Perhaps add more details.

Reply:
This is the EXPECT_EQ semantic, if EXPECT_EQ failed, it will print the `StringView(out, expectedMaxStringSize)`, I will drop it.

Reply:
Perhaps better to keep it for easy debugging but use std::string_view following
https://github.com/facebookincubator/velox/pull/10477#discussion_r1694966622.